hack/test-extended/default.sh

```diff
@@ -0,0 +1,224 @@
+MASTER_ADDR="${API_SCHEME}://${API_HOST}:${API_PORT}"
+PUBLIC_MASTER_HOST="${PUBLIC_MASTER_HOST:-${API_HOST}}"
+KUBELET_SCHEME="${KUBELET_SCHEME:-https}"
+KUBELET_HOST="${KUBELET_HOST:-127.0.0.1}"
+KUBELET_PORT="${KUBELET_PORT:-10250}"
+
+SERVER_CONFIG_DIR="${BASETMPDIR}/openshift.local.config"
+MASTER_CONFIG_DIR="${SERVER_CONFIG_DIR}/master"
+NODE_CONFIG_DIR="${SERVER_CONFIG_DIR}/node-${KUBELET_HOST}"
+
+# use the docker bridge ip address until there is a good way to get the auto-selected address from master
+# this address is considered stable
+# used as a resolve IP to test routing
+CONTAINER_ACCESSIBLE_API_HOST="${CONTAINER_ACCESSIBLE_API_HOST:-172.17.42.1}"
+
+STI_CONFIG_FILE="${LOG_DIR}/stiAppConfig.json"
+DOCKER_CONFIG_FILE="${LOG_DIR}/dockerAppConfig.json"
+CUSTOM_CONFIG_FILE="${LOG_DIR}/customAppConfig.json"
+GO_OUT="${OS_ROOT}/_output/local/go/bin"
+
+# set path so OpenShift is available
+export PATH="${GO_OUT}:${PATH}"
+
+cleanup() {
+    set +e
+    pid=$(cat ${BASETMPDIR}/server.pid 2>/dev/null)
+    if [ ! -z "$pid" ]; then
+      server_pids=$(pgrep -P $pid)
+      kill $server_pids $(cat ${BASETMPDIR}/server.pid) ${ETCD_PID}
+    fi
+    rm -rf ${ETCD_DIR-}
+
+	echo "[INFO] Stopping k8s docker containers"; docker ps | awk 'index($NF,"k8s_")==1 { print $1 }' | xargs -l -r docker stop
+	if [[ -z "${SKIP_IMAGE_CLEANUP-}" ]]; then
+		echo "[INFO] Removing k8s docker containers"; docker ps -a | awk 'index($NF,"k8s_")==1 { print $1 }' | xargs -l -r docker rm
+	fi
+
+	echo "[INFO] Removing ${BASETMPDIR}"
+	rm -rf ${BASETMPDIR}
+	if [[ $? != 0 ]]; then
+		echo "[INFO] Unmounting volumes ..."
+		findmnt -lo TARGET | grep openshift-${OS_TEST_NAMESPACE} | xargs -r sudo umount
+		rm -rf ${BASETMPDIR}
+	fi
+
+	remove_tmp_dir
+    echo "[INFO] Cleanup complete"
+}
+
+remove_tmp_dir() {
+	rm -rf ${BASETMPDIR} &>/dev/null
+	if [[ $? != 0 ]]; then
+		echo "[INFO] Unmounting volumes ..."
+		findmnt -lo TARGET | grep openshift-extended-tests | xargs -r sudo umount
+		rm -rf ${BASETMPDIR}
+	fi
+}
+
+trap "exit" INT TERM
+trap "cleanup" EXIT
+
+# Setup
+stop_openshift_server
+echo "[INFO] `openshift version`"
+echo "[INFO] Server logs will be at:    ${LOG_DIR}/openshift.log"
+echo "[INFO] Test artifacts will be in: ${ARTIFACT_DIR}"
+echo "[INFO] Volumes dir is:            ${VOLUME_DIR}"
+echo "[INFO] Config dir is:             ${SERVER_CONFIG_DIR}"
+echo "[INFO] Using images:              ${USE_IMAGES}"
+
+# Start All-in-one server and wait for health
+echo "[INFO] Create certificates for the OpenShift server"
+# find the same IP that openshift start will bind to.  This allows access from pods that have to talk back to master
+ALL_IP_ADDRESSES=`ifconfig | grep "inet " | sed 's/adr://' | awk '{print $2}'`
+SERVER_HOSTNAME_LIST="${PUBLIC_MASTER_HOST},localhost"
+while read -r IP_ADDRESS
+do
+	SERVER_HOSTNAME_LIST="${SERVER_HOSTNAME_LIST},${IP_ADDRESS}"
+done <<< "${ALL_IP_ADDRESSES}"
+
+openshift admin ca create-master-certs \
```

Comment:
This block is common (or so similar I don't see the difference right off) to here, `test-cmd.sh`, and `test-end-to-end.sh`.  Can you refactor to dry it out?